### PR TITLE
Restricted access to clients by client_id (permissions table)

### PR DESCRIPTION
**Pull request by sofiarm21**
### **Issue #378**

**Description:** 

This pr contains the necessary changes to add a `client_id` column in the `Permissions table` to be able to control the permission level of the contributors. For now, the default permission is `write` access to the contributor that created the client

**Breakdown:** 

* The column was added via a new migration
* The `client_id` is of type Integer and could be nullable
* When a new client is created the permission is automatically created with the contributor_id and the `client_id`. The `contributor_id` is given via the cookies session, but in case we are adding the client directly from the Apollo server console, we have to pass the `contributor_id` as one of the parameters.

**Implementation proof:**

https://www.loom.com/share/980318d98496485e9dd768fc779df14b

**Note:**

We have to decide who's going to have `read` permission to the client, there are two options

1. All the contributors have `read` access to all the clients.
2. Only the contributors thas has `read` or `write` permission to at least one project owned by the client will have `read` access to that client. This can be a script that gets called every time a user logs in.

Once we decided which one of these will be I will open a new issue to implement it.

### CODE REVIEW

**Comment by sofiarm21** [comment]
closes #378

File: api/schema/types/ClientType.js [hunk 1]
```diff
@@ -16,11 +16,19 @@ module.exports = gql`
...
         email: String,
         currency: String,
+        contributor_id: Int
+    }
+
+    input ClientInput {
+        is_active: Boolean,
+        name: String,
+        email: String,
+        currency: String
```
Comment (otech47) [inline api/schema/types/ClientType.js:31]: I think we don't need 2 separate input types especially since there are no strictly required`!` fields

**Review by otech47 — CHANGES_REQUESTED** [review]

File: api/schema/types/ClientType.js (hunk 1 above)
Comment (sofiarm21) [reply api/schema/types/ClientType.js:32]: The CinetCreateInput actually should be requires, so I will leave both input objects but one of them will be required

File: api/schema/resolvers/ClientResolver.js [hunk 2]
```diff
@@ -49,10 +49,25 @@ module.exports = {
...
     Mutation: {
-        createClient: (root, { createFields }, { models }) => {
-            return models.Client.create({
+        createClient: async (root, { createFields }, { cookies, models }) => {
+            const newlyCreatedClient = await models.Client.create({
                 ...createFields
             })
+            const contributor = (
+                await models.Contributor.findByPk(
+                    cookies ? cookies.userSession : createFields.client_id
```
[the remark this answers is not in the document]
Comment (sofiarm21) [reply api/schema/resolvers/ClientResolver.js]: fixed using `try catch` and making a validation before creating the client

**Comment by sofiarm21** [comment]
@otech47 change srequested resolved

**Pull request merged**